src/ports/postgres/modules/utilities/pivot.sql_in

```diff
@@ -0,0 +1,202 @@
+    pivot_cols.	The column name for the pivot is set as 
+    <em>'pivot name'</em>_<em>'pivot value'</em>.
+    </dd>
+    <dt>index </dt>
+    <dd>VARCHAR. Comma-separated columns that will form the index of the output 
+    pivot table.</dd>
+    <dt>pivot_cols </dt>
+    <dd>VARCHAR. Comma-separated columns that will form the columns of the 
+    output pivot table.</dd>
+    <dt>pivot_values </dt>
+    <dd>VARCHAR. Comma-separated columns that contain the values to be 
+    summarized in the output pivot table.</dd>
+
+</dl>
+
+@anchor notes
+@par Notes
+
+The default aggregate function is "sum". 
+
+NULL values in the index column are treated as any other value. 
+
+NULL values in the pivot column are ignored.
+
+NULL values in the value column are handled by the aggregate function.
+
+The following features are planned but not yet implemented.
+
+- Multiple index columns.
+- Multiple pivot columns.
+- Multiple value columns.
+- Aggregate functions as input.
+- NULL values in the pivot.
+
+
+@anchor examples
+@examp
+
+-#  Create a toy dataset.
+<pre class="example">
+CREATE TABLE pivset(
+                  id INTEGER,
+                  piv FLOAT8,
+                  val FLOAT8
+                );
+INSERT INTO pivset VALUES
+	(0, 10, 1),
+	(0, 10, 2),
+	(0, 20, 3),
+	(1, 20, 4),
+	(1, 30, 5),
+	(1, 30, 6),
+	(1, 10, 7),
+	(NULL, 10, 8),
+	(1, NULL, 9),
+	(1, 10, NULL);
+</pre>
+
+-# Pivot the table
+<pre class="example">
+DROP TABLE IF EXISTS pivout;
+SELECT madlib.pivot('pivset', 'pivout', 'id', 'piv', 'val');
+SELECT * FROM pivout;
+</pre>
+<pre class="result">
+ id | piv_10.0 | piv_20.0 | piv_30.0
+------+------------+------------+------------
+  0 |        3 |        3 |        
+  1 |        7 |        4 |       11
+    |        8 |        0 |        0
+</pre>
+*/
+
+-------------------------------------------------------------------------
+
+
+/**
+ * @brief Helper function that can be used to pivot tables
+ *
+ * @param source_table     The original data table
+ * @param out_table        The output table that contains the dummy
+ *                         variable columns
+ * @param index            The index columns to group by the records by
+ * @param pivot_cols       The columns to pivot the table
+ * @param pivot_values     The value columns to be summarized in the
+ *                         pivoted table
+ * @param aggregate_func   The aggregate function to be applied to the
+ *                         values
+ *
+ * @return Void
+ *
+ */
+ 
+CREATE OR REPLACE FUNCTION MADLIB_SCHEMA.pivot(
+    source_table        TEXT,
+    out_table           TEXT,
+    index               TEXT,
+    pivot_cols          TEXT,
+    pivot_values        TEXT,
+    aggregate_func      TEXT
+
+) RETURNS VOID AS $$
+    PythonFunction(utilities, pivot, pivot)
+$$ LANGUAGE plpythonu
+m4_ifdef(`__HAS_FUNCTION_PROPERTIES__', `MODIFIES SQL DATA', `');
+
+CREATE OR REPLACE FUNCTION MADLIB_SCHEMA.pivot(
```

Comment:
Instead of overloading the functions, it would be clearer to use default values for the parameters.

Though, it might be best to keep the "help" version separate; so add a default value for aggregate_func above, and then have a single argument version where the default value for the first argument is 'help'.